server/http.go

```diff
@@ -411,42 +692,67 @@ func (p *Plugin) handleStartMeeting(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	client, _, err := p.getActiveClient(user)
-	if err != nil {
-		p.API.LogWarn("Error getting the client", "err", err)
-		return
+	topic := req.Topic
+	if topic == "" {
+		topic = DefaultMeetingTopic
 	}
 
-	meeting, err := client.CreateMeeting(zoomUser, defaultMeetingTopic)
-	if err != nil {
-		p.API.LogWarn("Error creating the meeting", "err", err)
+	var meetingID int
+	var createMeetingErr error = nil
+	userPMISettingPref, getUserPMISettingErr := p.getPMISettingData(user.Id)
+	if getUserPMISettingErr != nil {
+		p.askUserForMeetingPreference(user.Id, req.ChannelID)
 		return
 	}
 
-	meetingID := meeting.ID
-	if err = p.postMeeting(user, meetingID, req.ChannelID, "", req.Topic); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+	switch userPMISettingPref {
+	case zoomPMISettingValueAsk:
+		p.askUserForMeetingPreference(user.Id, req.ChannelID)
 		return
+	case "", trueString:
+		meetingID = zoomUser.Pmi
+	default:
+		meetingID, createMeetingErr = p.createMeetingWithoutPMI(user, zoomUser, req.ChannelID, topic)
 	}
 
-	p.trackMeetingStart(userID, telemetryStartSourceWebapp)
+	if meetingID != -1 && createMeetingErr == nil {
```

Comment:
Should we log `createMeetingErr` if it's not nil?

Reply:
Yes, now I logged it.